detect_secrets/audit/report.py

```diff
@@ -0,0 +1,137 @@
+    for filename, secret_type, secret_hash, is_secret in details:
+        secret_class = SecretClass.from_boolean(is_secret)
+        if class_to_print != None and SecretClassToPrint.from_class(secret_class) != class_to_print:
+            continue
+        try:
+            detections = get_potential_secrets(filename, plugins[secret_type](), secret_hash)
+        except:
+            continue
+        identifier = hashlib.sha512((secret_hash + filename).encode('utf-8')).hexdigest()
+        for detection in detections:
+            if identifier in secrets:
+                secrets[identifier]['lines'][detection.line_number] = get_line_content(filename, detection.line_number)
+                if not secret_type in secrets[identifier]['types']:
+                    secrets[identifier]['types'].append(secret_type)
+                secrets[identifier]['class'] = secret_class.get_prioritary(secrets[identifier]['class']).to_string()
+            else:
+                finding = {}
+                finding['secret'] = detection.secret_value
+                finding['filename'] = filename
+                finding['lines'] = {}
+                finding['lines'][detection.line_number] = get_line_content(filename, detection.line_number)
+                finding['types'] = [secret_type]
+                finding['class'] = secret_class.to_string()
+                secrets[identifier] = finding
+
+    output = []
+    for identifier in secrets:
+        output.append(secrets[identifier])
+
+    print_message(json.dumps(output, indent=4, sort_keys=True))
+
+    
+def get_secrets_details_from_baseline(
+    baseline: str
+) -> [(str, str, str, bool)]:
+    """
+    :returns: Details of each secret present in the baseline file.
+    """
+    for filename, secrets in baseline['results'].items():
+        for secret in secrets:
+            yield filename, secret['type'], secret['hashed_secret'], secret['is_secret']
+
+
+def get_secret_class(
+    is_secret: bool
+) -> str:
+    """
+    :returns: Secret class as string.
+    """
+    return 'Unknown' if is_secret == None else 'True positive' if is_secret else 'False positive'
+
+
+def get_potential_secrets(
+    filename: str,
+    plugin: Plugin,
+    secret_to_find: str
+) -> [PotentialSecret]:
+    """
+    :returns: List of PotentialSecrets detected by a specific plugin in a file.
+    """
+    for lines in _get_lines_from_file(filename):
+        for line_number, line in list(enumerate(lines, 1)):
+            secrets = _scan_line(plugin, filename, line, line_number)
+            for secret in secrets:
+                if secret.secret_hash == secret_to_find:
+                    yield secret
+
+
+def get_line_content(
+    filename: str,
+    line_number: int
+) -> str:
+    """
+    :returns: Line content from filename by line number.
+    """
+    content = codecs.open(filename, encoding='utf-8').read()
+    if not content:
+        return None
+    return content.splitlines()[line_number - 1]
```

Comment:
There's actually already prior art for this, in `detect_secrets.audit.common.get_raw_secret_from_file`. Let's reuse this instead, as there are optimizations and edge case handling already built into that.

It may need to be refactored a little, but that should (hopefully) be pretty straightforward. e.g.

```python
def get_raw_secret_from_file(secret: PotentialSecret) -> str:
    line = get_raw_secret_line_from_file(secret)
    ...


def get_raw_secret_line_from_file(secret: PotentialSecret) -> str:
    ...
```

Reply:
I have some doubts about use the function `get_raw_secret_file` in the reporting feature.

I have reviewed the function `get_raw_secret_from_file` and I have seen that returns only the secret value in plaintext and only his first occurrence. I think that could be very interesting the inclusion of all the occurrences of one secret in the report, so I had implemented the `get_potential_secrets` function to get all of them.

Moreover, I have seen that the function `get_raw_secret_from_file` is based in the field `line_number` of the `PotentialSecret` (in this case, it is obtained from the baseline file), so if the baseline file is not updated when the user generates the report, this line number could be invalid and this function will raise the `SecretNotFoundOnSpecifiedLineError` exception. 

I think that the approach that I implemented can get all the required details about the secrets to generate a complete report. 

What do you think? Maybe could we implement a similar function in the `audit.common.py` file that obtain all the ocurrences of a secret? I will wait to know your opinion before apply any modification. 

Thank you very much!

Reply:
I'm also working in the report generation test, but maybe the resolution of this comment could have implications in the test implementation. I will push the test implementation after the resolution of this comment.

Thank you!